### PR TITLE
FIX: Updates Turris Greylist parser and feed URL.

### LINKED ISSUE #2167: Turris greylist has been updated
It seems that this morning the Turris graylist has been updated. The CSV file location has been changed from `https://project.turris.cz/greylist-data/greylist-latest.csv` to `https://view.sentinel.turris.cz/greylist-data/greylist-latest.csv`.

`http_url` parameter in `bots.collectors.http.collector_http` bot should be changed to `https://view.sentinel.turris.cz/greylist-data/greylist-latest.csv` to receive the correct input for the parser bot.

They also removed IP lookup data from the CSV file. (CSV header: `Address,Country,Tags,ASN` -> `Address,Tags`)
So you would also need to modify the `bots.parsers.turris.parser` file and remove `source.geolocation.cc` and `source.asn` from the `columns` list.

### PR DESCRIPTION
**Pull request by gethvi**
Fixes #2167.

### CODE REVIEW

File: intelmq/bots/parsers/turris/parser.py [hunk 1]
```diff
@@ -1,50 +1,49 @@
...
-        raw_report = utils.base64_decode(report.get("raw"))
-        raw_report = raw_report.translate({0: None})
-        for row in csv.reader(io.StringIO(raw_report)):
-            # ignore headers
-            if headers:
-                headers = False
-                continue
+    parse = ParserBot.parse_csv
+    recover_line = ParserBot.recover_line_csv
+    _ignore_lines_starting = ["#", "Address"]
```
Comment (sebix) [inline intelmq/bots/parsers/turris/parser.py]: is the line starting with `Address` the header? If yes, you may also use `parse_csv_dict` which parses the header and uses it as field names for the rows

File: intelmq/bots/parsers/turris/parser.py [hunk 2]
```diff
@@ -1,50 +1,49 @@
...
-import io
-
-from intelmq.lib import utils
 from intelmq.lib.bot import ParserBot
 
+PORTS = {
+    "ftp": 21,
+    "telnet": 23,
+    "http": 80
+    # smtp uses both 25 and 587, therefore we can't say for certain
```
Comment (sebix) [inline intelmq/bots/parsers/turris/parser.py:11]: is it guaranteed that the port can't be 990 for FTP over TLS (FTPS) or 443 for HTTP over TLS (HTTPS)?

File: intelmq/bots/parsers/turris/parser.py [hunk 3]
```diff
@@ -1,50 +1,49 @@
...
-            for key, value in zip(columns, row):
-                if key == "__IGNORE__":
-                    continue
+            if tag in ["smtp", "http", "ftp", "telnet"]:
+                event.add("protocol.transport", "tcp")
+                event.add("protocol.application", tag)
+                event.add("classification.type", "brute-force")
+
+                if PORTS.get(tag):
+                    event.add("destination.port", PORTS[tag])
```
Comment (sebix) [inline intelmq/bots/parsers/turris/parser.py]:
```suggestion
                event.add("destination.port", PORTS.get(tag))
```

File: intelmq/etc/feeds.yaml [hunk 4]
```diff
@@ -651,7 +651,7 @@ providers:
         collector:
           module: intelmq.bots.collectors.http.collector_http
           parameters:
-            http_url: https://www.turris.cz/greylist-data/greylist-latest.csv
+            http_url: https://view.sentinel.turris.cz/greylist-data/greylist-latest.csv
```
Comment (sebix) [inline intelmq/etc/feeds.yaml:654]: Please update the field `revision` of this entry too

**Review by sebix — CHANGES_REQUESTED** [review]

File: intelmq/bots/parsers/turris/parser.py (hunk 1 above)
Comment (gethvi) [reply intelmq/bots/parsers/turris/parser.py]: Thanks! Updated.

File: intelmq/bots/parsers/turris/parser.py (hunk 2 above)
Comment (gethvi) [reply intelmq/bots/parsers/turris/parser.py:11]: Yes. The data is based on Turris Sentinel Mini Honeypots, it uses only these ports. No TLS.

File: intelmq/etc/feeds.yaml (hunk 4 above)
Comment (gethvi) [reply intelmq/etc/feeds.yaml:654]: Updated.

**Review by sebix — APPROVED** [review]

File: intelmq/lib/upgrades.py [hunk 5]
```diff
@@ -887,7 +900,7 @@ def v310_feed_changes(configuration, harmonization, dry_run, **kwargs):
     ((3, 0, 1), (v301_deprecations,)),
     ((3, 0, 2), ()),
     ((3, 1, 0), (v310_feed_changes, v310_shadowserver_feednames)),
-    ((3, 2, 0), ()),
+    ((3, 2, 0), (v320_update_turris_greylist_url)),
```
Comment (sebix) [inline intelmq/lib/upgrades.py:903]: That line misses a trailing comma to make it a tuple:

```
    ((3, 2, 0), (v320_update_turris_greylist_url, )),
```

```
Upgrading to version 3.2.0.
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/intelmq/bin/intelmqsetup.py", line 184, in intelmqsetup_core
    controller.upgrade_conf(state_file=state_file, no_backup=True)
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/intelmq/bin/intelmqctl.py", line 1152, in upgrade_conf
    for function in bunch:
TypeError: 'function' object is not iterable
```

Comment (gethvi) [reply intelmq/lib/upgrades.py:903]: Fix of the fix incoming in few...

**Pull request merged**